code_generation/license.h

```diff
@@ -83,6 +87,31 @@ class KODE_EXPORT License
      */
     QString text() const;
 
+    /**
+     * Returns type of license.
+     */
+    Type type() const;
+
+    /**
+     * Return license name.
+     * @return name of license
+     */
+    QString typeName() const;
+
+    /**
+     * Returns license Type element by name. If there no such license,
+     * NoLicense will be returned
+     * @param license_string Name of license
+     * @return one of Type elements
+     */
+    static License licenseByTypeName(const QString &typeName);
+
+    /**
+     * Returns list of supported licenses
+     * @return vector of license names
+     */
+    static QStringList getSupportedLicenses();
```

Comment:
I don't get it, why should this return "internal" strings?
Are you showing this to a user?

If one day we add the enum value LGPL2dot1OrLater we won't want to show that exact string to users. We'll want to add a `QString displayName() const` to return something like "LGPL v2.1-or-later".

But then should this method return internal type names or display names? Tricky heh, since it depends on how it's going to be used. The typesafe solution is to return a QVector<License> or a QVector<Type> instead. Then the caller can transform that into a list of internal type names or a list of display names as desired.

Reply:
Maybe I need more details about the intended use case.

Reply:
kxml_compiler has feature to define license, but strings has been hardcoded. Here how it's look:
https://github.com/cornelius/kode/blob/master/kxml_compiler/kxml_compiler.cpp#L94:
```
    QCommandLineOption licenseOption(
            "license",
            QCoreApplication::translate(
                    "main", "License of generated files. Possible values: gpl, bsd, lgpl"),
            "license");
```
Here I'll change it to:
```
QCommandLineOption licenseOption(
            "license",
            QCoreApplication::translate("main", "License of generated files. Possible values: ")
                    + KODE::License::getSupportedLicenses().join(", "),
            "license");
```

And this https://github.com/cornelius/kode/blob/master/kxml_compiler/kxml_compiler.cpp#L280 
```
    if (cmdLine.isSet("license")) {
        QString l = cmdLine.value("license");
        if (l == "gpl") {
            c.setLicense(KODE::License(KODE::License::GPL));
        } else if (l == "bsd") {
            c.setLicense(KODE::License(KODE::License::BSD));
        } else if (l == "lgpl") {
            c.setLicense(KODE::License(KODE::License::LGPL));
        }
    }
```
I'm going replace with
```
    if (cmdLine.isSet("license")) {
        c.setLicense(KODE::License::licenseByTypeName(cmdLine.value("license")));
    }
```

So license handling will be unhardcoded and new license types can be added without updating logic in kxml_compiler.

Reply:
OK I see, thanks for the explanation. Indeed this is user-facing but command-line so spaces (and translations) aren't wanted. An "internal-looking" name is good for this indeed. OK.